pkg/standalone-utils/test/VaultActions.test.ts

```diff
@@ -608,4 +640,342 @@ describe('VaultActions', function () {
+                receipt,
+                vault.instance.interface,
+                'InternalBalanceChanged',
+                {
+                  user: sender.address,
+                  token: tokens.MKR.address,
+                }
+              );
+
+              daiAmountOut = daiTransfer.args.delta;
+              mkrAmountOut = mkrTransfer.args.delta;
+            } else {
+              const daiTransfer = expectEvent.inIndirectReceipt(
+                receipt,
+                new Interface((await getArtifact('v2-solidity-utils/ERC20')).abi),
+                'Transfer',
+                { from: vault.address, to: sender.address },
+                tokens.DAI.address
+              );
+              const mkrTransfer = expectEvent.inIndirectReceipt(
+                receipt,
+                new Interface((await getArtifact('v2-solidity-utils/ERC20')).abi),
+                'Transfer',
+                { from: vault.address, to: sender.address },
+                tokens.MKR.address
+              );
+
+              daiAmountOut = daiTransfer.args.value;
+              mkrAmountOut = mkrTransfer.args.value;
+            }
+
+            await expectChainedReferenceContents(toChainedReference(0), daiAmountOut);
+            await expectChainedReferenceContents(toChainedReference(1), mkrAmountOut);
+          });
+
+          it('exits with exact bpt in chained reference', async () => {
+            await setChainedReferenceContents(toChainedReference(0), amountInBPT);
+
+            await expectBalanceChange(
+              async () =>
+                relayer.connect(sender).multicall([
+                  await encodeExitPool({
+                    poolId: poolIdA,
+                    userData: WeightedPoolEncoder.exitExactBPTInForTokensOut(toChainedReference(0)),
+                    toInternalBalance,
+                  }),
+                ]),
+              await getBPT(poolIdA),
+              {
+                account: sender,
+                changes: {
+                  BPT: amountInBPT.mul(-1),
+                },
+              }
+            );
+          });
+
+          it('is chainable with swaps via multicall', async () => {
+            const receipt = await (
+              await expectBalanceChange(
+                async () =>
+                  relayer.connect(sender).multicall([
+                    await encodeExitPool({
+                      poolId: poolIdA,
+                      userData: WeightedPoolEncoder.exitExactBPTInForTokensOut(amountInBPT),
+                      toInternalBalance,
+                      outputReferences: {
+                        MKR: toChainedReference(0),
+                      },
+                    }),
+                    encodeSwap({
+                      poolId: poolIdA,
+                      tokenIn: tokens.MKR,
+                      tokenOut: tokens.DAI,
+                      fromInternalBalance: toInternalBalance,
```

Comment:
🤪

Reply:
Maybe rename `toInternalBalance` to `useInternalBalance` so that this doesn't look like an error?